internal/ctxt/merge_test.go

```diff
@@ -0,0 +1,129 @@
+package ctxt_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gophercloud/gophercloud/internal/ctxt"
+)
+
+func TestMerge(t *testing.T) {
+	t.Run("returns values from both parents", func(t *testing.T) {
+		ctx, cancel := ctxt.Merge(
+			context.WithValue(context.Background(), "key1", "value1"),
+			context.WithValue(context.Background(), "key2", "value2"),
```

Comment:
Can you perhaps also check that the value from the first context takes precedence when the second context holds the same key? If this is a behavior that we're going to support, then we should have tests for it so that we avoid regressions.

Reply:
this is a good idea, thanks